File: Data_Structure_java/LinkedList/SinglyLinkedList.java
public class SinglyLinkedList<E>{
	private Node<E> head = null;
	private Node<E> tail = null;
	private int size = 0;
	public SinglyLinkedList(){} //constructs an initially empty list
	//access methods
	public int size() {return size;}
	public boolean isEmpty() {return size == 0;}
	public E first(){
		if (isEmpty()) return null;
		else return head.getElement();
	}
	public E last(){
		if (isEmpty()) return null;
		else return tail.getElement();
	}

	//update method
	public void addFirst(E e){
		head = new Node<>(e,head); //create and link a new node
		if (size == 0) tail = head;
		size++;
	}

	public void addLast(E e){
		Node<E> newest = new Node<>(e,null);
		if (isEmpty()) head = newest;
		else tail.setNext(newest);
		tail = newest;
		size ++;
	}

	public E removeFirst() {
		if (isEmpty()) return null;
		E answer = head.getElement();
		head = head.getNext();
		size--;
		if (size == 0) tail = null;
		return answer;
	}

	public void printlist(){
		Node<E> search = head;
		while (search != null){
			if (search == tail) {System.out.println(search.getElement());}
			else System.out.print(search.getElement()+"->");
			search = search.getNext();
		}
	}
}

File: Data_Structure_python/LinkedList/SinglyLinkedList.py
from Node import Node
class SinglyLinkedList(Node):
	def __init__(self):
		self.size = 0
		self.head = None
		self.tail = None
	def size(self): return self.size
	def isEmpty(self): return (self.size == 0)
	def first(self):
		if self.head == None:
			return None
		else: return head.getElement()
	def last(self):
		if self.tail == None:
			return None
		else: return tail.getElement()

	#real method
	def addFirst(self,data):
		self.head = Node(data,self.head)
		if (self.isEmpty()):self.tail = self.head
		self.size += 1

	def addLast(self,data):
		node = Node(data,None)
		if (self.isEmpty()):
			self.head = node
		else:
			self.tail.setNext(node)
		self.tail = node
		self.size += 1

	def removeFirst(self):
		if (self.isEmpty()): return None
		else:
			answer = self.head.getElement()
			self.head = self.head.getNext()
			self.size -= 1
			if (self.isEmpty()):
				self.tail = None	
			return answer

	def printList(self):
		node = self.head
		print("head-> ",end = ' ')
		while (node != None):
			print(node.getElement(),end = ' ')
			node = node.getNext()
		print()